crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1530,74 +1559,137 @@ impl CumulocityConverter {
             .and_then(|entity| C8yTopic::smartrest_response_topic(entity, &self.config.c8y_prefix))
             .ok_or_else(|| Error::UnknownEntity(target.to_string()))?;
 
-        match response.status() {
+        let messages = match command.status() {
             CommandStatus::Init | CommandStatus::Scheduled | CommandStatus::Unknown => {
                 // The command has not been processed yet
-                Ok(vec![])
+                vec![]
             }
             CommandStatus::Executing => {
                 let smartrest_set_operation_status =
                     set_operation_executing(CumulocitySupportedOperations::C8ySoftwareUpdate);
-                Ok(vec![MqttMessage::new(
-                    &topic,
-                    smartrest_set_operation_status,
-                )])
+                vec![MqttMessage::new(&topic, smartrest_set_operation_status)]
             }
             CommandStatus::Successful => {
                 let smartrest_set_operation =
                     succeed_operation_no_payload(CumulocitySupportedOperations::C8ySoftwareUpdate);
 
-                Ok(vec![
+                vec![
                     MqttMessage::new(&topic, smartrest_set_operation),
-                    response.clearing_message(&self.mqtt_schema),
+                    command.clearing_message(&self.mqtt_schema),
                     self.request_software_list(target),
-                ])
+                ]
             }
             CommandStatus::Failed { reason } => {
                 let smartrest_set_operation =
                     fail_operation(CumulocitySupportedOperations::C8ySoftwareUpdate, &reason);
 
-                Ok(vec![
+                vec![
                     MqttMessage::new(&topic, smartrest_set_operation),
-                    response.clearing_message(&self.mqtt_schema),
+                    command.clearing_message(&self.mqtt_schema),
                     self.request_software_list(target),
-                ])
+                ]
+            }
+        };
+
+        Ok((
+            messages,
+            Some(command.into_generic_command(&self.mqtt_schema)),
+        ))
+    }
+
+    pub async fn upload_operation_log(
```

Comment:
The code of this method is close to both [`CumulocityConverter::handle_fts_log_download_result`](https://github.com/thin-edge/thin-edge.io/blob/main/crates/extensions/c8y_mapper_ext/src/operations/log_upload.rs#L180) and [`CumulocityConverter::handle_fts_config_download_result`](https://github.com/thin-edge/thin-edge.io/blob/main/crates/extensions/c8y_mapper_ext/src/operations/config_snapshot.rs#L181) which are already really close to each other.

A single method must be used to upload logs, log excerpts and config snapshots. This would avoid user-confusing small differences in upload content description.